core/src/main/scala/org/apache/spark/storage/BlockManagerMasterEndpoint.scala

```diff
@@ -427,6 +446,15 @@ class BlockManagerMasterEndpoint(
       locations.remove(blockManagerId)
     }
 
+    if (storageLevel.useDisk && externalShuffleServiceEnabled) {
+      val externalShuffleServiceId = externalShuffleServiceIdOnHost(blockManagerId)
```

Comment:
Shouldn't you de-dupe this id object? Seems like for a large cached RDD you'd end up with a lot of identical objects in memory... I remember seeing similar logic to de-dupe the regular block manager ids somewhere.

Reply:
I think we shouldn't. As `externalShuffleServiceIdOnHost` calls the `apply` method of `BlockManagerId` object where we have a cache for BlockManagerId instances:

https://github.com/apache/spark/blob/3e7797aa02a48e653b3c167a62cba6512a1d3bcb/core/src/main/scala/org/apache/spark/storage/BlockManagerId.scala#L122-L127